ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/PullFilterNode.java

```diff
@@ -295,7 +297,23 @@ public Void visitComparisonExpression(
         final ComparisonExpression node,
         final Object context
     ) {
+      if (!isSingleColumnReference(node)) {
```

Comment:
Just occurred to me: if the `WHERE` clause is just ` 100 = 101` then we would still do a table scan instead of immediately return empty values?